File: ImageJ/macros/Speed_Test/Speed_Test.js
  importClass(Packages.ij.IJ);
  importClass(Packages.ij.process.FloatProcessor);
  importClass(Packages.ij.ImagePlus);
  importClass(java.lang.System);
  
  sphere();
  
  // This code runs faster when it is in a function. 
  // The Nashorn JavaScript engine included with
  // Java 8 runs this script 27 times faster, compared
  // with the Rhino JavaScript used with Java 6, when
  // 'size' is set to 2048. Nashorn compiles JavaScript
  // into bytecode, which is JIT compiled into machine code.

  // With size =  4096 0.3 sec in Javascript!
  function sphere() {
     var size = 4096;
     var ip = new FloatProcessor(size,size);
     var t0 = System.currentTimeMillis();
     var x, y, dx, dy, d;
     for (y=0; y<size; y++) {
        for (x=0; x<size; x++) {
           dx=x-size/2;
           dy=y-size/2;
           d = Math.sqrt(dx*dx+dy*dy);
           ip.setf(x,y,-d);
        }
     }
     var time = (System.currentTimeMillis()-t0)/1000+" seconds";
     var img = new ImagePlus(time,ip);
     IJ.run(img,"Red/Green","");
     img.show();
  }

File: ImageJ/macros/Speed_Test/Speed_Test.py
"""
Sphere.py

A speed test example fixed

                     Modifications
  Date      Who  Ver                       What
----------  --- ------  -------------------------------------------------
2017-04-02  JRM 0.1.00  Initial test with 4096. Original example did not
                        import functions. 11.594sec on jrmFastMac
"""

from org.python.core import codecs
codecs.setDefaultEncoding('utf-8')

from java.lang import System
from ij import IJ, ImagePlus
from ij.process import FloatProcessor
from math import sqrt


t0 = System.currentTimeMillis()
size = 4096
ip = FloatProcessor(size,size)
for y in range(size):
   IJ.showProgress(y,size-1)
   for x in range(size):
       dx=x-size/2; dy=y-size/2
       d = sqrt(dx*dx+dy*dy)
       ip.setf(x,y,-d)
time = str((System.currentTimeMillis()-t0)/1000.0)+" seconds"
ImagePlus(time,ip).show()
IJ.run("Red/Green");